### ISSUE TITLE
Cache directory XML option ignored

### ISSUE THREAD

**Issue by MaurizioMoreo** [pestphp/pest#827]
The `cacheDirectory` option in the PHPUnit XML configuration file is ignored.

**Comment by nunomaduro** [comment]
Can you try to understand what's happening here?

**Comment by devajmeireles** [comment]
Hey, @MaurizioMoreo ! Thanks for your report.

First of all, I'll try to reproduce the same issue. If the same behavior happens with me, so I'll check if the Pest team if we have plans to do something related with that. Please, wait my feedback.

**Comment by krencl** [comment]
Hi, I figure out where the problem is:
In Application::run() is merged configuration from CLI arguments and XML configuration.
CLI arguments have higher priority than XML configuration. So CLI argument is used if is not null.
And because `Pest/Plugin`s (in this case `\Pest\Plugins\Cache`) is always injecting CLI arguments, there is never nullable CLI value and XML value is never used.

"Stack trace" to merger:
```
vendor/phpunit/phpunit/src/TextUI/Configuration/Merger.php:69
vendor/phpunit/phpunit/src/TextUI/Configuration/Registry.php:98
vendor/phpunit/phpunit/src/TextUI/Application.php:97
src/Kernel.php:91
bin/pest:91
```

"Stack trace" to injecting args from plugins:
```
src/Plugins/Actions/CallsHandleArguments.php:29
src/Kernel.php:88
bin/pest:91
```

**Comment by nunomaduro** [comment]
Fixed on `v2.33.0`
